src/game_options.c

```diff
@@ -0,0 +1,126 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "data_021BF67C.h"
+
+#include "struct_decls/struct_021C0794_decl.h"
+
+#include "struct_defs/game_options.h"
+
+#include "heap.h"
+#include "unk_02025E08.h"
+#include "game_options.h"
+
+GameOptions * GameOptions_Init (u32 heapID)
+{
+    GameOptions * newOptions = Heap_AllocFromHeap(heapID, sizeof(GameOptions));
+    GameOptions_SetDefaultValues(newOptions);
+
+    return newOptions;
+}
+
+void GameOptions_Copy (const GameOptions * src, GameOptions * dst)
+{
+    MI_CpuCopy8(src, dst, sizeof(GameOptions));
+}
+
+void GameOptions_SetDefaultValues (GameOptions * options)
+{
+    MI_CpuFill8(options, 0, sizeof(GameOptions));
+
+    options->textSpeed      = TEXTSPEED_MID;
+    options->audioMode      = AUDIOMODE_STEREO;
+    options->battleScene    = BATTLESCENE_ON;
+    options->battleStyle    = BATTLESTYLE_SHIFT;
+    options->buttonMode     = BUTTONMODE_NORMAL;
+    options->frameStyle     = FRAMESTYLE_01;
+}
+
+void GameOptions_UpdateGlobalButtonMode (UnkStruct_021C0794 * saveData, ButtonMode newButtonMode)
```

Comment:
Do we know if this is `newButtonMode` for a fact, even though `options->buttonMode` isn't set?